prov/verbs/src/fi_verbs.c

```diff
@@ -970,7 +1006,10 @@ static int fi_ibv_getinfo(uint32_t version, const char *node, const char *servic
 		goto err2;
 
 	if (hints) {
-		ret = fi_ibv_check_hints(hints, verbs_info);
+		/* TODO if we do not bind to a specific device we have to check 
+		 * check_info list and return those fi_info structures that match hints
+		 */
```

Comment:
Quoting your response:
> The verbs provider still supports listening across all underlying devices.  In that case, an fi_info structure would not be associated with a specific device.  Whether fi_info is associated with a device will depend on whether a source address has been selected/returned.

Yes, this is what I'm trying to convey in the comment. Since the app doesn't bind to a specific device, I think we should return a list of fi_info structures to the user. Each of the fi_info will correspond to a device. The app would pass this along when calling fi_passive_ep. There's a TODO in fi_ibv_eq_cm_getinfo that says "This should copy the listening fi_info as the base". So the fi_info passed to passive_ep should be used as the base, right?